lib/improver/nowcasting/lightning.py

```diff
@@ -0,0 +1,459 @@
+        #        increasing first-guess lightning probability to risk 1 (LR1).
+        #        This gives a decreasing influence on the extrapolated
+        #        lightning nowcast over forecast_period while retaining an
+        #        influence from the 50 km halo.
+        self.lrt_lev1 = lambda mins: 0.5 + mins * 2. / 360.
+        # Lightning-rate threshold for Lightning Risk 2 level
+        #        lrt_lev2 is the lightning rate threshold (as float) for
+        #        increasing first-guess lightning probability to risk 2 (LR2).
+        self.lrt_lev2 = 0.
+
+        # Set values for handling precipitation rate data
+        #    precipthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip).
+        #        These are the three prob(precip) thresholds and are designed
+        #        to prevent a large probability of lightning being output if
+        #        the probability of precipitation is very low.
+        self.precipthr = (0.0, 0.05, 0.1)
+        #    ltngthr (tuple):
+        #        Values for limiting prob(lightning) with prob(precip)
+        #        These are the three prob(lightning) values to scale to.
+        self.ltngthr = (0.0067, 0.25, 1.)
+        #    probability thresholds for increasing the prob(lightning)
+        #        phighthresh for heavy precip (>7mm/hr)
+        #            relates to problightning_values[2]
+        #        ptorrthresh for intense precip (>35mm/hr)
+        #            relates to problightning_values[1]
+        self.phighthresh = 0.4
+        self.ptorrthresh = 0.2
+
+        # Set values for handling vertically-integrated-ice (VII) data
+        #    ice_thresholds (tuple):
+        #        Values for increasing prob(lightning) with column-ice data.
+        #        These are the three VII thresholds in kg/m2.
+        self.ice_thresholds = (0.5, 1.0, 2.0)
+        #    ice_scaling (tuple):
+        #        Values for increasing prob(lightning) with VII data.
+        #        These are the three prob(lightning) values to scale to.
+        self.ice_scaling = (0.1, 0.5, 0.9)
+
+    def __repr__(self):
+        """
+        Docstring to describe the repr, which should return a
+        printable representation of the object.
+        """
+        return """<NowcastLightning: radius={radius},
+ lightning mapping (lightning rate in "min^-1"):
+   upper: lightning rate {lthru} => min lightning prob {lprobu}
+   lower: lightning rate {lthrl} => min lightning prob {lprobl}
+>""".format(radius=self.radius,
+            lthru=self.lrt_lev1, lthrl=self.lrt_lev2,
+            lprobu=self.pl_dict[1], lprobl=self.pl_dict[2])
+
+    def _update_metadata(self, cube):
+        """
+        Modify the meta data of input cube to resemble a Nowcast of lightning
+        probability.
+        1. Rename to "lightning_probability"
+        2. Remove "threshold" coord
+             (or causes iris.exceptions.CoordinateNotFoundError)
```

Comment:
I have a feeling there may be another metadata discussion pending here.  Without preempting that, some immediate thoughts:

- Inheriting some attributes (eg grid type, grid version and domain) from the precip cube is fine
- Inheriting attributes such as "title" and "source" might be questionable
- We should probably strip out cell methods explicitly
- Should the new cube name match the "first guess" lightning cube, ie "probability_of_lightning"?

Reply:
I think inheriting title and source is fine too. If they want to be different to the definitions in the precipitation cube, then this should be handled by the CLI instead.
I agree with cell-methods, change incoming...
I agree with the cube-name too (how did I miss that myself?)